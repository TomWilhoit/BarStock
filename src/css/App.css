/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE: 
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  /* background: #1d2023; */
}

body {
  height: 100%;
}

/* #endRegion Global */



.Header-container {
  padding-bottom: 2em;
}

.Footer-container {
  position: relative;
  width: 100%;
  text-align: center;
  /* background: #1d2023; */
}

