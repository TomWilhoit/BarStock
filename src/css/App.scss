/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE:
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  background: url(./dust_scratches.png);
}

body {
  height: 100%;
}

/* #endRegion Global */

#root {
  height: 100%;
}

.App {
  height: 100%;
}

.All-Content {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}


.Header-container {
  margin-bottom: 2em;
}


/* #region Content Container  */


.content-container {
  display: table;
  max-width: 1200px;
  width: 100%;
  margin: auto;
}

.Inventory-container {
  display: table-cell;
  width: 60%;
  padding: 0em 0em 0em 2em;
}

.Totals-container {
  display: table-cell;
  width: 40%;
  padding: 0em 2em 0em 2em;
  vertical-align: bottom;
}

/* #endRegion Content Container */


.Submit-order {
  button {
    width: 100%;
    border-radius: 4px;
    outline: none;
    background: #EEAD0E;
    color: #fff;
    text-transform: uppercase;
    font-weight: 100;
    font-size: 1em;
    padding: .2em;
    font-family: 'Abel', sans-serif;
    cursor: pointer;
    &.disabled {
      pointer-events: none;
      opacity: .2;
    }
    &.enabled {
      pointer-events: all;
      opacity: 1;
    }
  }
}


.Footer-container {
  width: 100%;
  text-align: center;
  margin-top: 2em;
}



@media only screen and (max-width: 800px) {
  .content-container {
    display: block;
  }
  .Inventory-container {
    margin: auto;
    display: block;
    width: 100%;
    max-width: 600px;
    padding: 0;
    padding-bottom: 2em;
  }
  .Totals-container {
    display: block;
    margin: auto;
    width: 100%;
    padding: 0;
    max-width: 600px;
  }
}
