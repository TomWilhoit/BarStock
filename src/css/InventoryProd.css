/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

*********************/

.Category-products {
    position: relative;
}

.Single-product.header {
    text-transform: uppercase;
    font-weight: bold;
    font-size: 1em;
    margin: .2em;
    padding: .2em;  
}

.Single-product.header span {
    /* color: #4e4e4e; */
    color: #7d6c54
}

.Single-product {
    margin: .5em 0 0 0;
    padding: .5em;
    /* border-bottom: solid 1px #231f20; */
    border-bottom: solid 1px #F8DE9E;
    display: -webkit-flex;
    display: flex;
    -webkit-justify-content: space-between;
    justify-content: space-between;
    -webkit-align-items: center;
    align-items: center;
    font-size: .85em;
    color: #7d6c54;
}

.Single-product span {
    margin: 0;
    width: 33%;
    font-family: 'Open Sans', sans-serif;
}
.Single-product span:first-child {
    text-align: left;
    font-size: 1em;
    letter-spacing: .03em;
}
.Single-product span:nth-child(2) {
    text-align: right;
}
.Single-product span:last-child {
    text-align: right;
}

.Single-product:last-child {
    border:none;
    margin: .5em 0 .5em 0;
}

.icon {
    color: #EEAD0E;
}