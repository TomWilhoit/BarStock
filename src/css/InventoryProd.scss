$font-copy: 'Lato', sans-serif;
$font-subHeader: 'Raleway', sans-serif;
$dark-blue: #11212a;
$light-blue: #0098de;
$light-gray: #adadad;
$white: #fff;

.Category-products {
  position: relative;
  overflow-y: scroll;
  max-height: 233px;

  .Single-product {
    padding: .5em .75em;
    border-bottom: solid 1px #ececec;
    display: flex;
    justify-content: space-between;
    font-family: $font-subHeader;

    &:nth-of-type(2n) {
      background:#f7f7f7cc;
    }

    &:last-child {
      border:none;
    }

    .product-img {
      max-width: 50px;
      line-height: 0;
      margin-right: 1.2em;
      img {
        width:100%;
      }
    }

    .product-info {
      flex-grow: 1;
      align-self: center;

      h4 {
        line-height: 1;
        padding-bottom: .4em;
        margin: 0;
        font-family: $font-copy;
        color: $dark-blue;
        font-weight: 300;
        font-size: 1em
      }

      h5 {
        line-height: 1;
        margin: 0;
        font-weight: 100;
        color: $light-gray;
        font-family: $font-subHeader;
        font-style: italic;

        span {
          font-size: 1.25em;
          padding-left: .2em;
          font-style: normal;
          font-weight: 100;
          color: $light-gray;
        }
      }
    }

    .product-quantity {
      align-self: center;
      width: 25%;
      display: flex;
      flex-direction: column;
      align-items: flex-end;

      .quantity-amount {
        display: flex;
        font-family: $font-copy;
        align-items: center;
        width: 100%;
        justify-content: space-around;

        .amount {
          font-size: 1.5em;
          color: $dark-blue;
        }

        button {
          border: none;
          background: $light-blue;
          outline: none;
          color: $white;
          border-radius: 4px;
          padding: 5px;
          line-height: 1;
          cursor: pointer;
          width:20%
        }
      }

      .quantity-label {
        text-align: center;
        width: 100%;

        p {
          margin: 0;
          color: $light-gray;
          font-family: $font-subHeader;
          font-size: .7em;
        }

        @media only screen and (max-width: 520px) {
          p {
            font-size: .5em;
            padding-top: .5em;
          }
        }
      }
    }

    @media only screen and (max-width: 520px) {
      padding: .5em .5em;

       .product-img {
         margin-right: .6em;
       }
       .product-quantity {
         width: 33%;
         .quantity-amount {
           button {
             width: 25%;
           }
         }
       }
       .product-info h4 {
         font-size: .8em;
       }
       .product-info h5 span {
         font-size: 1em;
       }
    }

    @media only screen and (max-width: 340px) {
      padding: .5em .5em;
      .product-info h5 span {
        font-size: .7em;
        font-weight: bold;
      }
    }

  }
}
