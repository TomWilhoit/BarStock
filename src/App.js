import React, { Component } from 'react';
import MockData from './mockData.js'
import './css/App.css';
import Inventory from './Inventory.js';
import Totals from './Totals.js';
import Header from './Header.js';
import Footer from './Footer.js';
import Login from './Login.js';



class App extends Component {
  constructor() {
    super();
    this.state = {
      allProducts: [MockData.distributor[0].inventory], 
      allMenu: [MockData.menu],
      cartItems: [],
      totalCost: 0,
      totalProjected: 0,
      currentUser: ""
    }
  }
  toggleAlert() {
    alert('toggle');
  }
  render() {
    return (
      <div className="App">
        {/* <div className="Login-container">
          <Login/>
        </div> */}
        <div className="Header-container">
          <Header/>
        </div>
        <div className="Inventory-container">
          <Inventory allProducts={this.state.allProducts}/>
        </div>
        <div className="Totals-container">
          <Totals allProducts={this.state.allProducts} 
                  allMenu={this.state.allMenu} />
        </div>
        <div className="Footer-container">
          <Footer/>
        </div>
      </div>
    )
  }
}

export default App;
