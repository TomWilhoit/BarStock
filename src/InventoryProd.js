import React, { Component } from 'react';
import App from './App.js';
import Inventory from './Inventory.js';
import InventoryCats from './InventoryCat.js';
import './css/InventoryProd.css';


class InventoryProd extends Component {
    constructor(props) {
        super(props);
        this.state = {
            // newProps: props
        }
    }
    
    
    render() {
        // console.log(this.state.newProps)
        //IS THIS THE RIGHT WAY TO BIND??
        let boundProps = this.props;
        return (
            <div>
            {
            boundProps.allProducts.filter(function (inventory) { 
                return inventory.category === boundProps.productType;
            }).map(function (el) {  
                return <p key={el.inventory_code}>{el.product}</p>
            })
            }
            </div>
        )
    }
}

export default InventoryProd;
