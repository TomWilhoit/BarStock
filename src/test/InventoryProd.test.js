import React from "react";
import App from "../App";
import { shallow } from "enzyme";
import InventoryProd from "../InventoryProd";
import mockData from "../mockData";

let products = mockData.distributor[0].inventory;
let category = mockData.distributor[0].inventory[0].category;
const changeCart = jest.fn();

describe("InventoryProd", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <InventoryProd
        allProducts={products}
        category={category}
        changeCart={changeCart}
      />
    );
  });
  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
