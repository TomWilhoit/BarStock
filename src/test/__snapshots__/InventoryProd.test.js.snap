// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryProd should match snapshot when all data is passed correctly 1`] = `
<article
  className="Category-products"
>
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10001"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10001,
        "price": 18.78,
        "product": "Fireball",
        "size": 33.8,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10002"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10002,
        "price": 16.99,
        "product": "Jim Beam",
        "size": 25.3,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10003"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10003,
        "price": 28.8,
        "product": "Jack Daniels",
        "size": 33.8,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10004"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10004,
        "price": 30.3,
        "product": "Bullit",
        "size": 33.8,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10005"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10005,
        "price": 31.88,
        "product": "Jameson",
        "size": 33.8,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10006"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10006,
        "price": 28.57,
        "product": "Leopold",
        "size": 33.8,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10007"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10007,
        "price": 36.71,
        "product": "Makers Mark",
        "size": 33.8,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10008"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10008,
        "price": 23.7,
        "product": "Tincup",
        "size": 33.8,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10009"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10009,
        "price": 8.21,
        "product": "Kentucky Deluxe",
        "size": 33.8,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
  <CartItem
    changeCart={[MockFunction]}
    className="CartItem"
    key="10010"
    product={
      Object {
        "category": "whiskey",
        "inventory_code": 10010,
        "price": 8.21,
        "product": "McCormicks Vodka",
        "size": 33.8,
        "type": "liquor",
        "unit": "ounces",
      }
    }
  />
</article>
`;
